### ISSUE TITLE
Add some status text to the log after a resource is created: "WW of XX are created, ZZ in progress, AA to be created"

### ISSUE THREAD

**Issue by afisher-stelligent** [hashicorp/terraform#31794]
### Terraform Version

```shell
$ terraform version                                                                           
Terraform v1.2.9
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.30.0
+ provider registry.terraform.io/hashicorp/random v3.4.3
```

### Use Cases

Developing new plans, it's nice to see what's been created/updated, what is still in flight, and what is not yet started. It really helps when working with provider constructs that you're not familiar with during a long apply. So when a resource is created, I'd love to see something like:

`aws_lb_listener.ss_alb_listener: Creation complete after 0s [id=arn:aws:elasticloadbalancing:us-west-2:663579103723:listener/app/ssLoadBalancer/374334f39fdfd045/cc6a66ded5350ad6]. 20 of 50 resources created, 10 still creating, 20 to create`

without having to turn on more verbose logging. Especially when testing in a pipeline like Jenkins, AWS CodePipeline, or GitHub Actions

### Attempted Solutions

There might be something in the debug logs, but it would be nice if, when a resource is created, what the progress of the plan is without having to turn on more verbose logging, cluttering up the logs

**Comment by apparentlymart** [comment]
Hi @afisher-stelligent! Thanks for this feature request.

For long running operations in Terraform today I would expect it to print "Still creating..." or "Still updating..." messages to tell you exactly what's currently in progress. Are you seeing those also in your output? If so, can you add a little more about what you're hoping to know that this current output doesn't tell you?

Thanks!

**Comment by afisher-stelligent** [comment]
Right... I do see all the "still creating/updating" messages, what I was hoping for was just something that summarized the progress of the overall apply after a resource is created... I thought the simplest would be something like:
`aws_lb_listener.ss_alb_listener: Creation complete after 0s [id=arn:aws:elasticloadbalancing:<stuff>:listener/app/ssLoadBalancer/374334f39fdfd045/cc6a66ded5350ad6]. 20 of 50 resources created, 10 still creating, 20 to create`

**Comment by apparentlymart** [comment]
Thanks for the extra context, @afisher-stelligent!

The way I understand your request is that you would prefer to have Terraform print a single-line, in-place-updating summary of progress instead of it emitting individual "still doing..." log lines for each individual object that is in progress.

Is it true that you are running Terraform in a terminal, which could therefore in principle support printing over the top of lines already written?

Terraform's current output is trying to strike a compromise by using terminal _formatting_ sequences but not cursor control sequences, because lots of folks run Terraform in situations where the output is not being processed by a full terminal emulator and so cursor control commands and overwriting previous output is ineffective.

However, I think we could consider the possibility of attempting to detect when the output is a terminal and, if so, adding an additional persistent status line with summary information like what you describe. I think the main requirement would be that Terraform not try to print it unless it can confirm that the standard output handle is connected to a terminal, so that the output would be unchanged for anyone who is running Terraform in a non-terminal context.
